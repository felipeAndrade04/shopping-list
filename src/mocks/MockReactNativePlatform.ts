jest.mock('react-native/Libraries/Utilities/Platform', () => {
  // eslint-disable-next-line @typescript-eslint/no-var-requires
  const fs = require('fs');
  const json = JSON.parse(fs.readFileSync('node_modules/react-native/package.json', 'utf8'));
  const platform = jest.requireActual('react-native/Libraries/Utilities/Platform');
  const version = json.version.toString().split('.');
  return {
    ...platform,
    constants: {
      ...platform.constants,
      reactNativeVersion: {
        major: version[0],
        minor: version[1],
        patch: version[2],
      },
    },
  };
});
