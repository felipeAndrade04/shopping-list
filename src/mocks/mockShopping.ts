import { Shopping } from '@app/models';

export const mockedShopping: Shopping = {
  products: [],
  name: 'Lista 8',
  created_at: new Date('01/01/2023'),
  id: '123',
};
