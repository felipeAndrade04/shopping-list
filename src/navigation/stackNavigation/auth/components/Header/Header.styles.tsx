import styled from 'styled-components/native';

export const Container = styled.View<{ height: number }>`
  height: ${({ height }) => height}px;
  padding-left: 16px;
  padding-bottom: 6px;
  justify-content: flex-end;
  background: ${({ theme }) => theme.colors.background};
`;
