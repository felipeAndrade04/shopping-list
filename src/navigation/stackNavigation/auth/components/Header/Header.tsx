import React from 'react';
import { Feather } from '@expo/vector-icons';
import { colors } from '@app/theme';
import { useSafeAreaInsets } from 'react-native-safe-area-context';
import * as S from './Header.styles';

interface Props {
  onPress: () => void;
}

export function Header({ onPress }: Props) {
  const insets = useSafeAreaInsets();

  const height = 90 + insets.bottom;
  return (
    <S.Container height={height}>
      <Feather
        testID="go-back"
        style={{ marginTop: 16 }}
        onPress={onPress}
        name="arrow-left"
        color={colors.dark}
        size={32}
      />
    </S.Container>
  );
}
