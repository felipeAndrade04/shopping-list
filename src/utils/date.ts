const months = ['jan', 'fev', 'mar', 'abr', 'mai', 'jun', 'jul', 'ago', 'set', 'out', 'nov', 'dez'];

export function formatDate(date: Date) {
  const day = date.getDay();
  const month = date.getMonth();

  return `${day} de ${months[month]}`;
}
