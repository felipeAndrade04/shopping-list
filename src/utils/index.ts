export * from './delay';
export * from './date';
