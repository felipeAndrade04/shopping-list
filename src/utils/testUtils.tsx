import React from 'react';
import { authReducer } from '@app/store';
import { ReactNode } from 'react';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components/native';
import * as theme from '@app/theme';
import { configureStore } from '@reduxjs/toolkit';

let store = configureStore({
  reducer: {
    auth: authReducer,
  },
});

export const wrapper = ({ children }: { children: ReactNode }) => (
  <ThemeProvider theme={theme}>
    <Provider store={store}>{children}</Provider>
  </ThemeProvider>
);

export function resetStore() {
  store = configureStore({
    reducer: {
      auth: authReducer,
    },
  });
}
