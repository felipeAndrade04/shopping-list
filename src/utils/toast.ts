import EventManager from './EvenManager';

export type ToastType = 'success' | 'info' | 'error';

export interface Toast {
  id?: string;
  type: ToastType;
  text: string;
  duration?: number;
}

export const toastEventManager = new EventManager();

export default function toast({ type, text, duration }: Toast) {
  toastEventManager.emit<Toast>('addtoast', {
    id: new Date().getTime().toString(),
    type,
    text,
    duration,
  });
}
