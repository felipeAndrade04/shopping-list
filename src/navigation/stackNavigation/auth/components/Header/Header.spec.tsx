import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import { wrapper } from '@app/utils';
import { Header } from './Header';
import { MockSafeAreaContext } from '@app/mocks';

describe('<Header />', () => {
  jest.mock('react-native-safe-area-context', () => MockSafeAreaContext);

  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('Should render correctly', () => {
    const onPress = jest.fn();
    const tree = render(<Header onPress={onPress} />, { wrapper }).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should make have go back', () => {
    const onPress = jest.fn();
    const { queryByTestId } = render(<Header onPress={onPress} />, {
      wrapper,
    });

    const button = queryByTestId('go-back');
    fireEvent.press(button);

    expect(onPress).toBeCalledTimes(1);
  });
});
