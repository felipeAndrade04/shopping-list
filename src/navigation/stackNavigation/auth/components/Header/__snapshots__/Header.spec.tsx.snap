// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Should render correctly 1`] = `
<View
  height={90}
  style={
    Array [
      Object {
        "backgroundColor": "#EEEEEE",
        "height": 90,
        "justifyContent": "flex-end",
        "paddingBottom": 6,
        "paddingLeft": 16,
      },
    ]
  }
>
  <RNGestureHandlerButton
    activeOpacity={0.105}
    collapsable={false}
    delayLongPress={600}
    handlerTag={1}
    handlerType="NativeViewGestureHandler"
    onActiveStateChange={[Function]}
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    onPress={[MockFunction]}
    style={
      Object {
        "marginTop": 16,
      }
    }
    testID="go-back"
    underlayColor="black"
  >
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "black",
          "borderBottomLeftRadius": undefined,
          "borderBottomRightRadius": undefined,
          "borderRadius": undefined,
          "borderTopLeftRadius": undefined,
          "borderTopRightRadius": undefined,
          "bottom": 0,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
    <Text />
  </RNGestureHandlerButton>
</View>
`;
